### DISCUSSION TITLE
How to display responsive images from a specific collection & conversion?

### DISCUSSION THREAD

**Discussion by cvlug** [spatie/laravel-medialibrary#2456] [Q&A]
My  (user)object has 2 conversions:
`    public function registerMediaConversions(Media $media = null): void
    {
        $this
            ->addMediaConversion('preview')
            ->optimize()
            ->height(100)
            ->nonQueued();

        $this
            ->addMediaConversion('optimized')
            ->optimize()
            ->height(650)
            ->withResponsiveImages()
            ->nonQueued();
    }`

My object olso has 2 collections, , see:
`    public function store(ProfileRequest $request)
    {
        $user = Auth::user();

        $user
            ->addFromMediaLibraryRequest($request->avatar)
            ->toMediaCollection('avatar');

        return back()->with('success', trans('Upload(s) geslaagd'));
    }

    public function storeMultiImages(ProfileMultiImagesRequest $request)
    {
        $user = Auth::user();

        $user
            ->syncFromMediaLibraryRequest($request->images)
            ->toMediaCollection('profileImages');

        return back()->with('success', trans(' multiple Upload(s) geslaagd'));
    }
`
When i try to display a responsive image with:
`{{ $user->getFirstMedia('optimized') }}`, nothing is outputted

I think is is because laravel-medialibrary does not know from  wich collection ('avatar' or 'profileImages') to display the media conversion.
What am i missing here?
How to fix?

**Comment by jam1e** [comment]
I'm aware this post is a little old now, but I came across it whilst trying to figure out the exact same thing. This is what seems to work for me (Using  v10.7):

`{{ $user->getFirstMedia('collection_name')->img('conversion_name') }}`
